packages/server/src/utils/ldap.ts

```diff
@@ -0,0 +1,74 @@
+import { Client } from 'ldapts';
+import { User } from '../services/database/types';
+import Logger from '@joplin/utils/Logger';
+import { LdapConfig } from './types';
+
+export default async function ldapLogin(email: string, password: string, user: User, config: LdapConfig): Promise<User> {
+	const logger = Logger.create('LDAP');
+
+	const enabled = config.enabled;
+	const userCreation = config.userCreation;
+	const host = config.host;
+	const mailAttribute = config.mailAttribute;
+	const fullNameAttribute = config.fullNameAttribute;
+	const baseDN = config.baseDN;
+	const bindDN = config.bindDN;
+	const bindPW = config.bindPW;
+
+	logger.info(`Starting authentication with Server ${host}`);
+
+	if (password === '') {
+		logger.error('no password entered');
+		throw new Error('no password entered');
```

Comment:
A regular error would be recorded as a server error (500), while this would be a user error (4xx). You could probably throw `ErrorForbidden` for example